### PR TITLE
yaml.load(s, Loader=...) requires Loader= argument always

### PR DESCRIPTION
**Pull request by ingydotnet**

### CODE REVIEW

**Review by perlpunk — COMMENTED** [review]
Maybe it would make sense to keep the warning for another while.
Note that `yaml.add_constructor` (and other methods) will add given constructors to `Loader`, `FullLoader` and `UnsafeLoader`, so people will be wondering why none of their custom constructors have an effect.
It should be recommended to only use `add_constructor` for a specific loader, and ideally to their own loader inheriting from one of the existing loaders.
While (hopefully) most users by now are using a specific loader for loading, there is still code out there that just disables the warning, and will stop working if they used custom constructors via `yaml.add_constructor`.
If something stops working and they enable warnings to investigate, they should be able to see a helpful message saying that the default loader has now been changed.

**Comment by ingydotnet** [comment]
> Maybe it would make sense to keep the warning for another while.

I'll add a warning if Full or Unsafe loaders had constructors added and load() is called without an explicit Loader=

**Comment by ingydotnet** [comment]
This started as `load()` defaults to `Loader=SafeLoader` but we all decided it was better to require `Loader=...` arg for `load()`.

**Review by Thom1729 — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Review by perlpunk — CHANGES_REQUESTED** [review]
Is there a test for the missing argument case?

File: tests/lib/test_dump_load.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import yaml
+
+def test_dump(verbose=False):
+    assert yaml.dump(['foo'])
+test_dump.unittest = True
+
+def test_load_safeloader(verbose=False):
```
Comment (nitzmahone) [inline tests/lib/test_dump_load.py:7]: Might want to add one that checks for `TypeError` if the second arg isn't specified?

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-.PHONY: default build buildext force forceext install installext test testext dist clean
```
Comment (nitzmahone) [inline Makefile:2]: is this unintended stuff for this PR?

**Comment by ingydotnet** [comment]
> Is there a test for the missing argument case?

Done in 1b2f534

File: Makefile (hunk 2 above)
Comment (ingydotnet) [reply Makefile:2]: Yes

File: tests/lib/test_dump_load.py (hunk 1 above)
Comment (ingydotnet) [reply tests/lib/test_dump_load.py:16]: Done in 1b2f534

**Review by nitzmahone — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**